fairscale/utils/parallel.py

```diff
@@ -89,28 +102,30 @@ def get_process_group_cached(ranks: Optional[Sequence[int]] = None) -> ProcessGr
             Return the requested process group. Throws RuntimeError if torch.distributed module is not yet initialized.
     """
     if not dist.is_initialized():
-        raise RuntimeError("torch.distributed is not yet initialized but process group is requested.")
+        # Likely caused by initiating a dummy pg for unit test, skip checking.
+        if name == ProcessGroupName.reduce_scatter and "pytest" in sys.modules:
+            return None
+        else:
+            raise RuntimeError("torch.distributed is not yet initialized but process group is requested.")
 
     # Init the cache if needed.
     if not hasattr(get_process_group_cached, "_global_group_cache"):
         get_process_group_cached._global_group_cache = {}  # type: ignore
         # Populate with default process group.
         cache = get_process_group_cached._global_group_cache  # type: ignore
-        assert dist.group.WORLD is not None
-        default_pg = dist.group.WORLD
-        if type(default_pg) == object:
-            # For PyTorch 1.6 and 1.7, dist.group.WORLD is an object, not a world process group, like that in 1.8 and 1.9.
-            default_pg = dist.new_group()
-        cache[None] = default_pg
-        cache[frozenset(list(range(dist.get_world_size())))] = default_pg
+
+        main_pg = dist.new_group(ranks=ranks)
+        cache[None] = main_pg
+        cache[(ProcessGroupName.main, None)] = main_pg
```

Comment:
curious but do you know why do we need three types of keys in the cache?

Reply:
When the rank is not specified, it goes to None.